src/js/util/object.js

```diff
@@ -0,0 +1,53 @@
+function isObject(something) {
+  return typeof something === 'object';
+}
+
+/**
+ *
+ * @param {Object} object
+ */
+function cloneDeepObject(object) {
+  const keys = Object.keys(object);
+  return Object.freeze(
+    keys.reduce((result, key) => {
+      const value = object[key];
+      if (Array.isArray(value)) {
+        return { ...result, [key]: [...value] };
+      }
+      if (typeof value === 'object') {
+        return { ...result, [key]: Object.freeze({ ...cloneDeepObject(value) }) };
+      }
+      return { ...result, [key]: value };
+    }, {})
+  );
+}
+
+/**
+ *
+ * @param {[]} array
+ */
+function cloneDeepArray(array) {
+  return array.reduce((result, item) => {
+    if (Array.isArray(item)) {
+      return [...result, ...cloneDeepArray(item)];
+    }
+    if (typeof value === 'object') {
+      return [...result, cloneDeepObject(item)];
+    }
+    return [...result, item];
+  }, []);
+}
```

Comment:
```suggestion
function cloneDeepObject(object) {
  return Object.entries(object)
            .reduce((result, [key, value]) => ({
              ...result, [key]: cloneDeep(value)
            }), {});
}

function cloneDeepArray(array) {
  return array.map(cloneDeep);
}
```
생각해보니, 이렇게 표현해도 되겠네요 ㅎㅎ

Reply:
부연 설명을 해드리자면, get 메서드를 이용해서 객체를 취득하면 해당 객체의 쓰기도 엄격히 금지시키고 싶은 의도가 섞여 있었습니다. 그래서 객체의 키에 해당하는 값이 'object'인 경우 Object.freeze를 사용해서 객체의 변경을 방어할 의도가 있었습니다. 제안해주신 코드가 명확하고 깔끔해보이지만, 검증을 위해 불가피하게 코드가 추가된 점에 관해서 말씀드리고자 합니다. 이 부분 관련해서는 복사만 하는 것이 아니니 메서드의 이름에 readOnly를 추가하도록 할게요!

또한, 이번 프로젝트에서는 객체를 복사해서 재사용할 의도는 없으므로 시간관계상 읽기전용만 가능한 복사된 객체를 만드는 함수로 작성하겠습니다.

Reply:
해당 부분은 좀 더 바람직한 방법을 찾았습니다.
객체 깊은 복사 -> 객체 깊은 Freezing -> 반환 

상수를 freezing해야 하기 때문에 freezing하는 과정은 별개로 분리하는 것이 타당하다는 판단이 들었습니다.
보완해서 리뷰 요청드릴게요!